platform/engineering/design-docs/2020-04-03-separate-content-build.md

```diff
@@ -0,0 +1,486 @@
+  build repo
+  
+#### React application landing pages
+JavaScript application landing pages will be created during the content build.
+
+**Phase 1: Stopgap solution**  
+To mirror the current process for creating application landing pages, we will
+modify the `createReactPages` Metalsmith plugin to use a JSON file with the
+parameters instead of pulling the properties from the various `manifest.json`
+files. This JSON file will include:
+- URL of the page
+- Application name
+- Whether the file should be made available in production
+
+When we do this, we will update the `manifest.json` documentation and remove
+unused properties from all the files.
+
+**Phase 2: Long-term solution**  
+All application landing pages will be created in the CMS. The page's published
+state will determine whether the application should be made available in
+production.
+
+Once this is complete, we will remove the Phase 1 stopgap solution.
+
+#### Cache busting application code
+To ensure the latest code is downloaded by the browser, we have to bust the
+browser's cache. **Currently**, we're adding hashes to the asset file names and
+changing the `<script>` tag's `src` property to point to the file name with
+hash.
+
+Once we split out the content build, however, it's infeasible to continue this
+approach. Instead, we'll rely on the (`Cache-Control`
+header)[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control].
+Setting it to `no-cache` will require browsers to check with the server for
+every request before using the cached assets.
+
+**Note:** This will cause higher traffic for S3 and may result in a lower
+performance for users, as the browser will have to make a request to the server
+before the cache can be used.
+
+### Code Location
+To start, the **content build** will live at
+`vets-website/src/stages/build-content/`. (This should be mostly just a rename
+of `build/` to `build-content/` for clarity.) Once the implementation is
+complete, this code will **move to a new repository.**
+
+The **application build** will consist of **Webpack configuration files** found
+in the **root of the project.** If we need any script files to aid in the build
+process, they will be located in `vets-website/scripts/build/`.
+
+### Testing Plan
+
+#### Consolidated build script
+After the content and application builds have their own build scripts within
+`vets-website`, we'll write a script to run both builds together to reproduce
+the current build script's output. To test that this is working properly we can:
+
+1. Run the script pre-separation (checkout `master` for this)
+1. Rename the `vets-website/build/localhost/` directory
+1. Run the post-separation build script
+1. Check that the all same files are there
+    - We can use `find` and `cmp` for this as needed
+    
+### Logging
+**Question:** What _should_ it log? Anything? Is this a Jenkins / Nomad /
+CircleCI thing?
+
+### Debugging
```

Comment:
We should mention the logs that will be available to troubleshoot branch and deployment builds. The primary UI is going to be in Circle- we're going to do as much of the process as we can there.